rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -37,25 +37,11 @@ __rmw_destroy_publisher(
   const rmw_node_t * node,
   rmw_publisher_t * publisher)
 {
-  if (!node) {
-    RMW_SET_ERROR_MSG("node handle is null");
-    return RMW_RET_ERROR;
-  }
-
-  if (node->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
-
-  if (!publisher) {
-    RMW_SET_ERROR_MSG("publisher handle is null");
-    return RMW_RET_ERROR;
-  }
-  if (publisher->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
+  assert(node->implementation_identifier == identifier);
```

Comment:
I think you need to add:

```cpp
(void)identifier;
```

If you want to avoid a warning in release mode.

Reply:
It's used below.